### PR TITLE
Actions for the MySQL charm

### PR DESCRIPTION
**Pull request by Abuelodelanada**

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -75,6 +87,60 @@ def _on_update_status(self, event):
...
+        creds = {
+            "username": event.params["username"],
+            "password": event.params["password"],
+            "hostname": "%",
+        }
+
+        try:
+            self.mysql.set_user_password(creds)
+            event.set_results({"username": creds["username"]})
+            event.log(f"Pasword fo username: {creds['username']} changed")
```
Comment (balbirthomas) [inline src/charm.py:117]: Password **for**

**Review by balbirthomas — APPROVED** [review]

File: src/charm.py [hunk 2]
```diff
@@ -75,6 +87,60 @@ def _on_update_status(self, event):
...
+    def _on_create_user_action(self, event):
+        """Handle the create_user action."""
+        creds = {
+            "username": event.params["username"],
+            "password": event.params["password"],
+            "hostname": "%",
+        }
+
+        try:
+            self.mysql.new_super_user(creds)
```
Comment (mmanciop) [inline src/charm.py:99]: Will this actually delete the previous user?

**Review by mmanciop — COMMENTED** [review]
It is unclear to me what happens to an old superuser after creating a new one

File: src/charm.py (hunk 2 above)
Comment (Abuelodelanada) [reply src/charm.py:99]: No, this action does not delete previous users.
This action verifies if the username is already taken. If so will show `"User already exists: {credentials['username']}"`

**Comment by Abuelodelanada** [comment]
> It is unclear to me what happens to an old superuser after creating a new one

At the end this action creates the new user and runs: `GRANT ALL PRIVILEGES ON *.* TO 'username'@'%'`

**Pull request merged**